receiver/dockerstatsreceiver/receiver.go

```diff
@@ -115,8 +120,260 @@ func (r *receiver) scrape(ctx context.Context) (pmetric.Metrics, error) {
+			"total":   r.mb.RecordContainerBlockioIoServiceBytesRecursiveTotalDataPoint,
+		}},
+		{entries: blkioStats.IoServiceTimeRecursive, opToRecorderMap: map[string]blkioRecorder{
+			"read":    r.mb.RecordContainerBlockioIoServiceTimeRecursiveReadDataPoint,
+			"write":   r.mb.RecordContainerBlockioIoServiceTimeRecursiveWriteDataPoint,
+			"sync":    r.mb.RecordContainerBlockioIoServiceTimeRecursiveSyncDataPoint,
+			"async":   r.mb.RecordContainerBlockioIoServiceTimeRecursiveAsyncDataPoint,
+			"discard": r.mb.RecordContainerBlockioIoServiceTimeRecursiveDiscardDataPoint,
+			"total":   r.mb.RecordContainerBlockioIoServiceTimeRecursiveTotalDataPoint,
+		}},
+		{entries: blkioStats.IoServicedRecursive, opToRecorderMap: map[string]blkioRecorder{
+			"read":    r.mb.RecordContainerBlockioIoServicedRecursiveReadDataPoint,
+			"write":   r.mb.RecordContainerBlockioIoServicedRecursiveWriteDataPoint,
+			"sync":    r.mb.RecordContainerBlockioIoServicedRecursiveSyncDataPoint,
+			"async":   r.mb.RecordContainerBlockioIoServicedRecursiveAsyncDataPoint,
+			"discard": r.mb.RecordContainerBlockioIoServicedRecursiveDiscardDataPoint,
+			"total":   r.mb.RecordContainerBlockioIoServicedRecursiveTotalDataPoint,
+		}},
+		{entries: blkioStats.IoTimeRecursive, opToRecorderMap: map[string]blkioRecorder{
+			"read":    r.mb.RecordContainerBlockioIoTimeRecursiveReadDataPoint,
+			"write":   r.mb.RecordContainerBlockioIoTimeRecursiveWriteDataPoint,
+			"sync":    r.mb.RecordContainerBlockioIoTimeRecursiveSyncDataPoint,
+			"async":   r.mb.RecordContainerBlockioIoTimeRecursiveAsyncDataPoint,
+			"discard": r.mb.RecordContainerBlockioIoTimeRecursiveDiscardDataPoint,
+			"total":   r.mb.RecordContainerBlockioIoTimeRecursiveTotalDataPoint,
+		}},
+		{entries: blkioStats.IoWaitTimeRecursive, opToRecorderMap: map[string]blkioRecorder{
+			"read":    r.mb.RecordContainerBlockioIoWaitTimeRecursiveReadDataPoint,
+			"write":   r.mb.RecordContainerBlockioIoWaitTimeRecursiveWriteDataPoint,
+			"sync":    r.mb.RecordContainerBlockioIoWaitTimeRecursiveSyncDataPoint,
+			"async":   r.mb.RecordContainerBlockioIoWaitTimeRecursiveAsyncDataPoint,
+			"discard": r.mb.RecordContainerBlockioIoWaitTimeRecursiveDiscardDataPoint,
+			"total":   r.mb.RecordContainerBlockioIoWaitTimeRecursiveTotalDataPoint,
+		}},
+		{entries: blkioStats.SectorsRecursive, opToRecorderMap: map[string]blkioRecorder{
+			"read":    r.mb.RecordContainerBlockioSectorsRecursiveReadDataPoint,
+			"write":   r.mb.RecordContainerBlockioSectorsRecursiveWriteDataPoint,
+			"sync":    r.mb.RecordContainerBlockioSectorsRecursiveSyncDataPoint,
+			"async":   r.mb.RecordContainerBlockioSectorsRecursiveAsyncDataPoint,
+			"discard": r.mb.RecordContainerBlockioSectorsRecursiveDiscardDataPoint,
+			"total":   r.mb.RecordContainerBlockioSectorsRecursiveTotalDataPoint,
+		}},
+	} {
+		for _, entry := range blkioRecorder.entries {
+			if recorder, ok := blkioRecorder.opToRecorderMap[strings.ToLower(entry.Op)]; ok {
```

Comment:
In the event that `entry.Op` is not present in the map, there should be a way to surface that information. 

An error may be the best means of doing so or providing some debug logs that can be configured on collector start up. Having things silently fail is worrisome.

Reply:
Yeah, good point. I think a debug log is appropriate. I will add one.

Reply:
Done